### PR TITLE
CLOUDSTACK-8592: Quota plugin usage documentation

### PR DESCRIPTION
**Pull request by abhinandanprateek**
This document provides information on configuring and running the quota service. It also provides information on various APIs to check and maintain quotas for accounts.

### CODE REVIEW

File: source/plugins/quota.rst [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+KIND, either express or implied.  See the License for the specific language
+governing permissions and limitations under the License.
+
+
+Quota Plugin 
+=============
+
+Quota service, while allowing for scalability, will make sure that the cloud is
+not exploited by attacks, careless use and program errors. To address this
+problem, we propose to employ a quota-enforcement service that allows resource
```
Comment (DaanHoogland) [inline source/plugins/quota.rst]: We don't propose anymore once this is out. "To address this problem, a quota-enforcement service can be employed. This service allows resource usage to be restricted to certain bounds as..."

File: source/plugins/quota.rst [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+usage within certain bounds as defined by policies and available quotas for
+various entities. Quota service extends the functionality of usage server to
+provide a measurement for the resources used by the accounts and domains using a
+common unit referred to as cloud currency in this document. It can be configured
+to ensure that your usage won’t exceed the budget allocated to accounts/domain
+in cloud currency. It will let user know how much of the cloud resources he is
+using. It will help the cloud admins, if they want, to ensure that a user does
+not go beyond his allocated quota. Per usage cycle if a account is found to be
```
Comment (DaanHoogland) [inline source/plugins/quota.rst]: typo: an account (not a account)

File: source/plugins/quota.rst [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+exceeding its quota then it is locked. Locking an account means that it will not
```
Comment (DaanHoogland) [inline source/plugins/quota.rst]: then it is locked => , it will be locked

File: source/plugins/quota.rst [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+By default Quota service does not lock the accounts that have exceeded the quota
+usage. To enable quota service to lock accounts set the following global
+configuration to true:
+
+#. quota.enable.enforcement
+
+The other configurations that are there for quota service are as:
+
+#. quota.currency.symbol : The symbal that is used before any currency
```
Comment (DaanHoogland) [inline source/plugins/quota.rst]: "The symbol that is used in front of any currency figure in various ...

File: source/plugins/quota.rst [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+	timeout duration.
+
+There are several configuration variables that are inherited from usage server.
+These are listed below:
+
+#. usage.aggregation.timezone 
+
+All these are described in details in Usage Server documentation.
+
+Restart the Management Server (as usual with any global configuration change)
```
Comment (DaanHoogland) [inline source/plugins/quota.rst]: usual but not always needed. I would rephrase to (to enable the set configuration values)

File: source/plugins/quota.rst [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+| 16               | CPU\_NUMBER                       | The quota tariff for    |
+|                  |                                   | using 1 virtual CPU.    |
++------------------+-----------------------------------+-------------------------+
+| 17               | MEMORY                            | The quota tariff for    |
+|                  |                                   | using 1MB RAM size.     |
++------------------+-----------------------------------+-------------------------+
+
+The quota tariff can be listed using listQuotaTariff API.
+
+quotaTariff: ListLists all quota tariff plans
```
Comment (DaanHoogland) [inline source/plugins/quota.rst]: ListLists is a typo?

**Comment by DaanHoogland** [comment]
remarks made but LGTM

**Comment by abhinandanprateek** [comment]
Daan, 
  Thanks for reviewing the document, the observations are incorporated.

**Comment by ke4qqq** [comment]
LGTM +1

**Comment by runseb** [comment]
LGTM +1

**Comment by runseb** [comment]
@pdion this should be on the master branch not in the 4.6 branch since quota have not yet been merged...correct ?

**Comment by pdion891** [comment]
hum, ok, I will roll it back from 4.6 docs.

**Comment by runseb** [comment]
@pdion891 let me check with @abhinandanprateek I might be wrong

**Comment by abhinandanprateek** [comment]
@runseb  @pdion891  This did not make it into 4.6, will be merged with master.

https://github.com/apache/cloudstack/pull/768

**Comment by pdion891** [comment]
thanks, I've looked into Jira, and rollback the publishing of quotas in 4.6 documentation. still merge in master, master branch is not published on RTD.  So we are all good I think.

Thanks for rising this @runseb

**Pull request closed without merging**